src/cascadia/LocalTests_TerminalApp/TabTests.cpp

```diff
@@ -1242,10 +1258,13 @@ namespace TerminalAppLocalTests
         END_TEST_METHOD_PROPERTIES()
 
         auto page = _commonSetup();
-        page->RenameWindowRequested([&page](auto&&, const winrt::TerminalApp::RenameWindowRequestedArgs args) {
+        page->RenameWindowRequested([&page, this](auto&&, const winrt::TerminalApp::RenameWindowRequestedArgs args) {
             // In the real terminal, this would bounce up to the monarch and
             // come back down. Instead, immediately call back and set the name.
-            page->WindowName(args.ProposedName());
+            //
+            // This replicates how TerminalWindow works
+            _windowProperties->WindowName(args.ProposedName());
+            page->WindowNameChanged();

```

Comment:
The decoupling of setting window name and notifying that it changed gives me pause. Will we ever mess it up? Can it be ensured that we don't?

Reply:
Yea I really wanted this to be observable on the WindowProperties, so that the page would just listen for the PropertyChanged. There was some reason that didn't work, lemme see if I can remember...